ReactiveCocoa/Swift/Property.swift

```diff
@@ -90,16 +90,13 @@ public final class MutableProperty<Value>: MutablePropertyType {
 	public var value: Value {
 		get {
 			lock.lock()
-			let value = _value
-			lock.unlock()
-			return value
+			defer { lock.unlock() }
+
+			return _value
 		}
 
 		set {
-			lock.lock()
-			_value = newValue
-			observer.sendNext(newValue)
-			lock.unlock()
+			modify { _ in newValue }
```

Comment:
`swap` is a syntactic sugar of `modify`, and in terms of `Atomic` it is used quite a few times, e.g. in Signal.swift there is `self.atomicObservers.swap(nil)`. At least it saves you to type `{ _ in }` if the intention is just swapping the value.